src/v/cloud_storage/remote.h

```diff
@@ -455,8 +452,7 @@ class remote : public ss::peering_sharded_service<remote> {
         }
 
     private:
-        std::optional<std::reference_wrapper<retry_chain_node>>
-          _source_to_ignore;
+        absl::node_hash_set<const retry_chain_node*> _sources_to_ignore;
```

Comment:
Nit: a named type instead of `const retry_chain_node*`, so it's clear that _sources_to_ignore is not meant to be used to reach a retry_chain_node object but only for an identity check.

and a comment over retry_chain_node deleted move constructor saying that the address is saved in _sources_to_ingnore

Reply:
Good idea. I'll do this in a follow-up.